sdk/metric/provider.go

```diff
@@ -42,7 +42,7 @@ func NewMeterProvider(options ...Option) *MeterProvider {
 	flush, sdown := conf.readerSignals()
 
 	mp := &MeterProvider{
-		pipes:      newPipelines(conf.res, conf.readers, conf.views, conf.exemplarFilter),
+		pipes:      newPipelines(conf.res, conf.readers, conf.views, conf.exemplarFilter, conf.cardinalityLimit),
```

Comment:
Is there any test that actually ensures that the cardinality limit is applied (in an more or less end to end way)?
I feel that we should add a `TestCardinalityLimit` below `TestExemplarFilter` in `meter_test.go`.

Reply:
I have not seen https://github.com/open-telemetry/opentelemetry-go/pull/7081#issuecomment-3138800692 when I started reviewing. I think it would be good to add such notice in the PR description.